gtars/src/uniwig/mod.rs

```diff
@@ -169,13 +186,22 @@ fn clamped_start_position(start: i32, smoothsize: i32) -> i32 {
 /// Main function
 pub fn uniwig_main(
     smoothsize: i32,
-    combinedbedpath: &str,
+    filepath: &str,
     chromsizerefpath: &str,
     bwfileheader: &str,
     output_type: &str,
+    filetype: &str,
 ) -> Result<(), Box<dyn Error>> {
-    let stepsize = 1;
+    // Determine File Type
+    let ft = match filetype.to_lowercase().as_str() {
```

Comment:
A slightly more idiomatic approach might be to implement `FromStr` for this:
```rust
use std::str::FromStr;
use std::fmt;

#[derive(Debug)]
enum FileType {
    BED,
    BAM,
}

impl FromStr for FileType {
    type Err = String;

    fn from_str(s: &str) -> Result<Self, Self::Err> {
        match s.to_lowercase().as_str() {
            "bed" => Ok(FileType::BED),
            "bam" => Ok(FileType::BAM),
            _ => Err(format!("Invalid file type: {}", s)),
        }
    }
}

// Usage
let ft = FileType::from_str("BED");
```

Reply:
Not sure where to put the enum and its impls, maybe in a `filetypes.rs` file?

Reply:
Oh this is neat. I implemented it! Thanks for the suggestion!